packages/gestalt/src/ButtonToggle.tsx

```diff
@@ -134,34 +163,115 @@ const ButtonToggleWithForwardRef = forwardRef<HTMLButtonElement, Props>(function
     [focusStyles.accessibilityOutline]: !disabled && isFocusVisible,
   });
 
+  // Consume GlobalEventsHandlerProvider
+  const { buttonToggleHandlers } = useGlobalEventsHandlerContext() ?? {
+    buttonToggleHandlers: undefined,
+  };
+
+  const bgColor: 'red' | 'transparent' = color instanceof Array ? 'red' : color;
+  if (color instanceof Array) {
+    return (
+      <button
+        ref={innerRef}
+        aria-controls={accessibilityControls}
+        aria-label={accessibilityLabel}
+        aria-pressed={selected}
+        className={classnames(sharedTypeClasses, styles.colorPickerButton, {
+          [styles.colorPickerButtonDisabled]: disabled,
+        })}
+        data-test-id={dataTestId}
+        disabled={disabled}
+        onBlur={(event) => {
+          onBlur?.({ event });
+        }}
+        onClick={(event) => {
+          buttonToggleHandlers?.onClick?.();
+          onClick?.({ event });
+        }}
+        onFocus={(event) => {
+          onFocus?.({ event });
+        }}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchCancel={handleTouchCancel}
+        onTouchEnd={handleTouchEnd}
+        // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
```

Comment:
@ts-expect-error - TS2322 should disappear we have a HTMLButtonElement now

Reply:
They don't though, because of the useTapFeedback hook that types tap targets as HTML Div Elements.
I am using this hook for consistency with Button - suggestions accepted